### PR TITLE
Performance Improvements for Observability Instrumentation

### PR DESCRIPTION
**Pull request by nadundesilva**

### CODE REVIEW

File: compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/bir/codegen/JvmObservabilityGen.java [hunk 1]
```diff
@@ -686,26 +676,22 @@ private void injectStopObservationCall(BIRBasicBlock observeEndBB, DiagnosticPos
...
-                                               DiagnosticPos pos) {
-        BIRVariableDcl constLoadVariableDcl = new BIRVariableDcl(constantType,
-                new Name(String.format("$%s$const", uniqueId)), VarScope.FUNCTION, VarKind.TEMP);
-        scopeVarList.add(constLoadVariableDcl);
-        BIROperand operand = new BIROperand(constLoadVariableDcl);
-        ConstantLoad constLoadIns = new ConstantLoad(pos, constantValue, constantType, operand);
-        basicBlock.instructions.add(constLoadIns);
-        return operand;
+    private BIROperand generateGlobalConstantOperand(BIRPackage pkg, BType constantType, Object constantValue) {
+        return compileTimeConstants.computeIfAbsent(constantValue, k -> {
```
Comment (Kishanthan) [inline compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/bir/codegen/JvmObservabilityGen.java:687]: Shall we just go with old style for loops?

Comment (Kishanthan) [reply compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/bir/codegen/JvmObservabilityGen.java:687]: Ignore

File: compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/bir/codegen/JvmObservabilityGen.java [hunk 2]
```diff
@@ -118,9 +126,10 @@ public JvmObservabilityGen(JvmPackageGen pkgGen) {
      * @param pkg The package to instrument
      */
     public void rewriteObservableFunctions(BIRPackage pkg) {
+        compileTimeConstants = new HashMap<>();
```
Comment (Kishanthan) [inline compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/bir/codegen/JvmObservabilityGen.java:129]: Can we move this initialization to constructor? Is this method called only once?

Comment (nadundesilva) [reply compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/bir/codegen/JvmObservabilityGen.java]: Fixed in 759b37d

**Pull request merged**